src/System.Management.Automation/engine/runtime/CompiledScriptBlock.cs

```diff
@@ -2043,15 +2048,19 @@ public static string Match(string text)
                         // iter n+1: compute hash on `mit` (len 3)
                         //    This overwrites the previous iteration, hence we go from longest to shortest.
                         //
-                        // LCG comes from a trivial (bad) random number generator,
+                        // Original code:
+                        // rh[j] = 31 * rh[j - 1] + h;
+                        // 31 comes from a trivial (bad) random number generator,
                         // but it's sufficient for us - the hashes for our patterns
                         // are unique, and processing of 2200 files found no false matches.
-                        runningHash[j] = LCG * runningHash[j - 1] + h;
+                        //
+                        // Then replace multiplication with more efficient operations.
+                        rh[j] = (rh[j - 1] << 5) - rh[j - 1] + h;
```

Comment:
Given that you have gone so far, maybe do the following change to avoid additional boundary checks?

```suggestion
                        uint v = rh[j - 1];
                        rh[j] = (v << 5) - v + h;
```

Reply:
We can if you want but compiler does the optimization.

Reply:
> The compiler knows how to replace multiplication by a power of two with a shift, but we do this explicitly to be safe

This is from your PR description. If so, then I'd prefer to not do anything, just rely on the compiler to do the right thing.
It's a micro-optimization anyway, which is likely won't make a difference in overall script execution performance.